### PR TITLE
fix: Use bitcoin regtest rather than bitcoin testnet

### PR DESCRIPTION
**Pull request by ielashi**
# Description

For local bitcoin development, the replica needs to be initialized with the `bitcoin_regtest` subnet feature and not `bitcoin_testnet`.

This fix depends on [MR 5394](https://gitlab.com/dfinity-lab/public/ic/-/merge_requests/5394) in the replica, and should not be merged until that MR is merged.

# How Has This Been Tested?

* Built `ic-starter` and `replica` from the branch of the MR above.
* Used `DFX_REPLICA_PATH` and `DFX_IC_STARTER_PATH` to use these locally built binaries.
* Started bitcoind and generated some blocks
* Cloned this dfx project: https://github.com/ielashi/bitcoin-integration-demo
* Ran my locally modified dfx with `dfx start --enable-bitcoin --bitcoin-node 127.0.0.1:18444`, and then verified that the adapter and replica were syncing blocks correctly from bitcoind.
* Deployed the canister in my dfx project.
* Verified that the canister can read the balances/UTXOs of the bitcoin state.

# Checklist:

- [x] The title of this PR complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/).
- [x] I have edited the CHANGELOG accordingly.

### CODE REVIEW

**Comment by ielashi** [comment]
@ericswanson-dfinity I'll be out for the next two weeks, would you mind merging this PR after reviewing it?

Note that some tests will currently fail. That's because this change requires us to update the replica version we're using. We need to wait for [this MR](https://gitlab.com/dfinity-lab/public/ic/-/merge_requests/5394) on the replica to be merged, then we can update to the latest replica for the tests to pass.

**Comment by dsarlis** [comment]
@ericswanson-dfinity The replica [MR](https://gitlab.com/dfinity-lab/public/ic/-/merge_requests/5394) we had been waiting on has been merged. Can you take the necessary steps to get this approved and merged please?

**Comment by ericswanson-dfinity** [comment]
> @ericswanson-dfinity The replica [MR](https://gitlab.com/dfinity-lab/public/ic/-/merge_requests/5394) we had been waiting on has been merged. Can you take the necessary steps to get this approved and merged please?

In addition to the MR being merged, we need the commit to make it into a replica version that has been elected and deployed to mainnet. Once that happens, yes.

**Comment by dsarlis** [comment]
> > @ericswanson-dfinity The replica [MR](https://gitlab.com/dfinity-lab/public/ic/-/merge_requests/5394) we had been waiting on has been merged. Can you take the necessary steps to get this approved and merged please?
> 
> In addition to the MR being merged, we need the commit to make it into a replica version that has been elected and deployed to mainnet. Once that happens, yes.

Ok, I didn't know that's the process. In this case, I'd expect this to happen next week (next RC cut will be next Wednesday and usually election happens on Friday or something like that).

File: CHANGELOG.adoc [hunk 1]
```diff
@@ -71,6 +71,11 @@ This incorporates the following executed proposals:
 
 == DFX
 
+=== fix: Initialize replica with bitcoin regtest flag
+
+When the bitcoin feature is enabled, dfx was launching the replica with the "bitcoin_testnet" feature.
+The correct feature to use is "bitcoin_regtest".
+
```
Comment (ericswanson-dfinity) [inline CHANGELOG.adoc:78]: This needs to go in the "unreleased" section above

**Review by ericswanson-dfinity — CHANGES_REQUESTED** [review]

File: CHANGELOG.adoc (hunk 1 above)
Comment (dsarlis) [reply CHANGELOG.adoc]: Done.

**Review by ericswanson-dfinity — APPROVED** [review]

File: src/dfx/src/actors/replica.rs [hunk 2]
```diff
@@ -309,7 +309,7 @@ fn replica_start_thread(
             cmd.args(&["--http-port", &port.to_string()]);
         }
         if config.btc_adapter.enabled {
-            cmd.args(&["--subnet-features", "bitcoin_testnet"]);
```
Comment (rumenov) [inline src/dfx/src/actors/replica.rs:312]: does bitcoin_regtest translates correctly into regtest string in the adapter config?

File: src/dfx/src/actors/replica.rs [hunk 3]
```diff
@@ -309,7 +309,7 @@ fn replica_start_thread(
...
+            cmd.args(&["--subnet-features", "bitcoin_regtest"]);
             if let Some(socket_path) = config.btc_adapter.socket_path {
                 cmd.args(&[
                     "--bitcoin-testnet-uds-path",
```
Comment (ericswanson-dfinity) [inline src/dfx/src/actors/replica.rs:315]: Also, note that the second parameter involved is `--bitcoin-testnet-uds-path`.  Should this be `--bitcoin-regtest-uds-path` (which does not currently exist as a parameter for ic-starter) ?

File: src/dfx/src/actors/replica.rs (hunk 2 above)
Comment (ielashi) [reply src/dfx/src/actors/replica.rs:312]: This was already hard-coded here: https://github.com/dfinity/sdk/blob/9a377bc74e48db4063789108f47f33700155aa9d/src/dfx/src/lib/bitcoin/adapter/config.rs#L58

File: src/dfx/src/actors/replica.rs (hunk 3 above)
Comment (ielashi) [reply src/dfx/src/actors/replica.rs:315]: @ericswanson-dfinity This is correct as-is. In theory, this uds-path is not specific to any particular network and should just be `--uds-path`. The networking team is aware of the inaccuracy of the current nomenclature and I expect they'll sort that out at some point. cc @rumenov

**Pull request merged**